File: src/main/java/me/viniciusarnhold/altaria/events/interfaces/IReceiver.java
package me.viniciusarnhold.altaria.events.interfaces;

import org.jetbrains.annotations.NotNull;
import sx.blah.discord.api.events.Event;

/**
 * Created by Vinicius.
 *
 * @since ${PROJECT_VERSION}
 */
public interface IReceiver {

    @NotNull
    Class<? extends Event> getEventType();

    void disable();

}

File: core/src/main/kotlin/me/viniciusarnhold/altaria/events/interfaces/IReceiver.kt
package me.viniciusarnhold.altaria.events.interfaces

import sx.blah.discord.api.events.Event

/**
 * Created by Vinicius.

 * @since ${PROJECT_VERSION}
 */
interface IReceiver {

    val eventType: Class<out Event>

    fun disable()

}
